packages/editor/src/browser/editor-manager.ts

```diff
@@ -64,10 +67,14 @@ export class EditorManagerImpl implements EditorManager, WidgetFactory {
         @inject(SelectionService) protected readonly selectionService: SelectionService,
         @inject(FrontendApplication) protected readonly app: FrontendApplication,
         @inject(WidgetManager) protected readonly widgetManager: WidgetManager,
-        @inject(FileIconProvider) protected readonly iconProvider: FileIconProvider
+        @inject(FileIconProvider) protected readonly iconProvider: FileIconProvider,
+        @inject(WorkspaceCommandContribution) protected readonly uiEvent: WorkspaceCommandContribution
     ) {
         this.currentObserver = new EditorManagerImpl.Observer('current', app);
         this.activeObserver = new EditorManagerImpl.Observer('active', app);
+        uiEvent.onIUriChanged(event => {
+            this.renameWidget(event.factoryId.toString(), event.options, event.newOptions, event.newLabel);
```

Comment:
We should get the FileMovedEvent from FS here.
And then figureout the proper way to rename based on the URI change. we have all the info needed.